crud/common/utils.lua

```diff
@@ -338,6 +354,44 @@ function utils.format_result(rows, space, field_names)
     return result
 end
 
+function utils.merge_comparison_fields(space_format, key_parts, field_names)
```

Comment:
This is not "merge" - we crop space_format to field_names and add adds `cmp_keys_parts`.

This function is also specific for select plan, so it can be moved to `select.plan`. 
Moreover, can it be refactored?

I mean, we can transform `field_names` and use it later.

```
local field_names = enrich_field_names_with_cmp_key(field_names, cmp_key_parts)
```

But `updated_space_metadata.key_parts` is used only by `gen_tuples_comparator`, so I suggest you go pass `field_names` to this function and take care about `updated_space_metadata.key_parts`  inside. And it would be great to add a comments why does `fields` optsion affects creating `tuples_comparator`.